### ISSUE TITLE
machine converge property - no longer used?

### ISSUE THREAD

**Issue by keen99** [chef/chef-provisioning#427]
https://docs.chef.io/resource_machine.html
````
converge	
Ruby Types: TrueClass, FalseClass
Use to manage convergence when used with the :create action. Set to false to prevent convergence. Set to true to force convergence. When nil, the machine will converge only if something changes. Default value: nil.
````

https://github.com/chef/chef-provisioning/blob/master/lib/chef/resource/machine.rb#L58-L60
https://github.com/chef/chef-provisioning/blob/master/lib/chef/resource/machine.rb#L58-L60
````
  # Allows you to turn convergence off in the :create action by writing "converge false"
  # or force it with "true"
  attribute :converge, :kind_of => [TrueClass, FalseClass]
````

As best I can tell from reading and testing, this property is no longer relevant - it only applied to the `action :create` which was removed in https://github.com/chef/chef-provisioning/commit/e42be06c7a814ef77acfe5c7e30ed0116b8019ea

Can we remove this and it's associated documentation to prevent confusion?  Or, if the intention is to have this work...update the docu and fix it so it works?

chef on the remote is definitely converged regardless of the state of this property (true/false/nil) with `action :converge`

Ran across this in an attempt to find a only-converge-once-ever option.  (because for me the runlist applied only bootstraps another provisioner...and while the recipe can handle running more than once, doing `:setup` and `:ready` and `:converge` every single time this gets run sucks up a lot of time....

**Comment by tyler-ball** [comment]
The converge property should still be respected, even if the action is `:converge`.  Marking this as a bug so we can investigate it.

```ruby
machine "foo" do
  action :converge
  converge false
end
```

The expected behavior of this recipe is that we will do everything to converge the machine _except_ issue the `chef-client` command.  So we will install chef, setup the keys, register the node with the server, etc.

This also brings up a very common use case that we don't have good support or documentation around - using `chef-provisioning` to provision a node while _only_ running chef-client once.  After that, the `chef-client` recipe should be managing the client run configuration.

**Comment by keen99** [comment]
> This also brings up a very common use case that we don't have good support or documentation around - using chef-provisioning to provision a node while only running chef-client once.

yes!  I -really- want a "converge only once. but DO converge once, even if it takes a few times" option.

it's on my todo list to hack in one way or another....

**Comment by tyler-ball** [comment]
I tested the `converge` attribute and found it being respected.  The problem is that the `install_sh.rb` logic triggers the [`resource_updated?`](https://github.com/chef/chef-provisioning/blob/master/lib/chef/provider/machine.rb#L65) check to return true every time.  I'm going to open a separate bug for that
